arq/connections.py

```diff
@@ -133,34 +144,58 @@ async def enqueue_job(
                 return
         return Job(job_id, redis=self, _queue_name=_queue_name, _deserializer=self.job_deserializer)
 
-    async def _get_job_result(self, key) -> JobResult:
-        job_id = key[len(result_key_prefix) :]
+    async def _get_job_result(self, key, partial: bool = False) -> JobResult:
```

Comment:
please can you explain the idea behind "partial" results - I think it's added since I last reviewed this PR?

Again, I think this might need to be in a separate PR.